app/services/recipe_generator_service.rb

```diff
@@ -41,8 +42,17 @@ def system_message
   end
 
   def prompt
+    preference_info = @preferences.map { |pref|
+      "Consider this preference: #{pref.description}. Restriction: #{pref.restriction}"
+    }.join("\n")
```

Comment:
What if you move this to a method? Also doing this join it makes this text to be at the top. I think is better to be after introducing prompt to be an assistant.

Reply:
I don't know if I understood the 'join' part correctly, so I will ask for review again in a moment.